AElf.Kernel/BlockBody.cs

```diff
@@ -7,32 +7,54 @@ namespace AElf.Kernel
     [Serializable]
     public class BlockBody : IBlockBody
     {
-        private List<ITransaction> Transactions { get; set; }
+        private List<ITransaction> _transactions { get; set; } = new List<ITransaction>();
+        private MerkleTree<IAccount> _stateMerkleTreeRightChild = new MerkleTree<IAccount>();
 
         public int TransactionsCount
         {
             get
             {
-                return Transactions.Count;
+                return _transactions.Count;
             }
         }
 
         public BlockBody() { }
 
         public IQueryable<ITransaction> GetTransactions()
         {
-            return Transactions.AsQueryable();
+            return _transactions.AsQueryable();
+        }
+
+        public MerkleTree<IAccount> GetChangedWorldState()
+        {
+            return _stateMerkleTreeRightChild;
         }
 
         public bool AddTransaction(ITransaction tx)
         {
             //Avoid duplication of addition.
-            if (Transactions.Exists(t => t.GetHash() == tx.GetHash()))
+            if (_transactions.Exists(t => t.GetHash() == tx.GetHash()))
             {
                 return false;
             }
-            Transactions.Add(tx);
+            _transactions.Add(tx);
             return true;
         }
+
+        /// <summary>
+        /// For one account, just update once its state.
+        /// </summary>
+        /// <param name="account"></param>
+        /// <returns></returns>
+        public bool AddState(IAccount account)
+        {
+            var hash = new Hash<IAccount>(ExtensionMethods.GetHash(account));
+            if (null == _stateMerkleTreeRightChild.FindLeaf(hash))
+            {
```

Comment:
It's only still but I think this equality test should be reversed : _stateMerkleTreeRightChild.FindLeaf(hash) == null

Reply:
N